src/Microsoft.SqlTools.ServiceLayer/Security/SecurityService.cs

```diff
@@ -169,6 +184,152 @@ await requestContext.SendResult(new ResultStatus()
+            prototype.IsLockedOut = login.IsLockedOut;
+            prototype.IsDisabled = !login.IsEnabled;
+            prototype.MustChange = login.EnforcePasswordPolicy ? login.MustChangePassword : false;
+            prototype.WindowsGrantAccess = login.ConnectPermission;
+
+            if (prototype.LoginType == SqlServer.Management.Smo.LoginType.SqlLogin)
+            {
+                // check that there is a password
+                // this check is made if policy enforcement is off
+                // with policy turned on we do not display this message, instead we let server
+                // return the error associated with null password (coming from policy) - see bug 124377
+                if (prototype.SqlPassword.Length == 0 && prototype.EnforcePolicy == false)
+                {
+                    // raise error here
+                }
+
+                // check that password and confirm password controls' text matches
+                if (0 != String.Compare(prototype.SqlPassword, prototype.SqlPasswordConfirm, StringComparison.Ordinal))
+                {
+                    // raise error here
+                }
+            }
+
+            var _ = prototype.ServerRoles.ServerRoleNames;
+            foreach (string role in login.ServerRoles)
+            {
+                prototype.ServerRoles.SetMember(role, true);
+            }
+
+            prototype.ApplyGeneralChanges(dataContainer.Server);
+            prototype.ApplyServerRoleChanges(dataContainer.Server);
+            prototype.ApplyDatabaseRoleChanges(dataContainer.Server);
+            await requestContext.SendResult(new object());
+        }
+
+        internal async Task HandleInitializeLoginViewRequest(InitializeLoginViewRequestParams parameters, RequestContext<LoginViewInfo> requestContext)
+        {
+            contextIdToConnectionUriMap.Add(parameters.ContextId, parameters.ConnectionUri);
+            ConnectionInfo connInfo;
+            ConnectionServiceInstance.TryFindConnection(parameters.ConnectionUri, out connInfo);
+            if (connInfo == null) 
+            {
+                // raise an error
+            }
+            CDataContainer dataContainer = CDataContainer.CreateDataContainer(connInfo, databaseExists: true);
+            LoginViewInfo loginViewInfo = new LoginViewInfo();
+
+            string[] databases = new string[dataContainer.Server.Databases.Count];
+            for (int i = 0; i < dataContainer.Server.Databases.Count; i++)
+            {
+                databases[i] = dataContainer.Server.Databases[i].Name;
+            }
+            string[] languages = new string[dataContainer.Server.Languages.Count];
+            for (int i = 0; i < dataContainer.Server.Languages.Count; i++)
+            {
+                languages[i] = dataContainer.Server.Languages[i].Name;
+            }
+
+            LoginPrototype prototype = parameters.IsNewObject 
```

Comment:
it might make sense to cache the initial prototype between requests for a dialog, since I suspect it takes several DB queries to populate.  We can see if that's worth it later.